### PR TITLE
Alerting: Add safeguard for migrations that might cause dataloss

### PR DESCRIPTION
**Pull request by JohnnyQQQQ**
This PR adds a safeguard if anyone tries to revert from unified alerting to legacy alerting. The safeguard is needed as some customers faced data loss after this migration and we need a second validation layer for this action.

### CODE REVIEW

**Review by santihernandezc — APPROVED** [review]
I would add:
- Some test cases with `ForceMigration = false`.
- An entry to the changelog.

We can add documentation for this new flag after this PR to solve the incident is merged.

**Comment by JacobsonMT** [comment]
LGTM! Some small things though:

- Please add an internal changelog entry
- Should this be added to docs or left hidden for now?
- I think adding a test case for "when unified alerting disabled and migration is already run but force_migration is false, then do nothing" would be appropriate here, something like the following with some sort of test for panic afterwards.

```golang
{
	name: "when unified alerting disabled and migration is already run but force_migration is false, then do nothing",
	config: &setting.Cfg{
		UnifiedAlerting: setting.UnifiedAlertingSettings{
			Enabled: boolPointer(false),
		},
		ForceMigration: false,
	},
	isMigrationRun: true,
	expected:       []string{},
	expectPanic:    true,
},
```

**Comment by JohnnyQQQQ** [comment]
Added the panic test and the missing documentation.

File: docs/sources/administration/configuration.md [hunk 1]
```diff
@@ -135,6 +135,10 @@ Options are `production` and `development`. Default is `production`. _Do not_ ch
 Set the name of the grafana-server instance. Used in logging, internal metrics, and clustering info. Defaults to: `${HOSTNAME}`, which will be replaced with
 environment variable `HOSTNAME`, if that is empty or does not exist Grafana will try to use system calls to get the machine name.
 
+### force_migration
+
+Force migration will run migrations that might cause dataloss. Default is `false`.
```
Comment (JohnnyQQQQ) [inline docs/sources/administration/configuration.md:140]:
```suggestion
Force migration will run migrations that might cause data loss. Default is `false`.
```

**Pull request merged**